src/nodes.coffee

```diff
@@ -1104,47 +1115,77 @@ exports.Obj = class Obj extends Base
 
   children: ['properties']
 
+  isAssignable: ->
+    @lhs = true
```

Comment:
This is a bit hacky, but I wasn't sure how else to deal with `{a=1}` being valid only when destructuring. The other approach I considered was adding a "destructuring"/"assignment"/"lhs" option to `o`.

```coffee
class Assign
  ...
  compileNode: (o) ->
    o.assignment = true
    ...

class Obj
  ...
  compileNode: (o) ->
    ...
    prop.operatorToken.error "unexpected #{prop.operatorToken.value}" unless o.assignment
```